### ISSUE TITLE
Does not work after reboot

### ISSUE THREAD

**Issue by simonsmartza** [googlesamples/google-services#6]
This works well when you run it on any handset from Android studio.
However when you reboot the device, it does not work any more

I tried ALL the solutions on Stackexchange but none worked 

Please help!!!

**Comment by silvolu** [comment]
Hi @simonsmartza, are you referring to any specific sample?

**Comment by simonsmartza** [comment]
Hi,
Thank you for your response.
Sorry, I am referring to the gcm sample. 
I install it as per documentation.
It work after install, but if you reboot the device the notifications are not picked up on some devices.
(i.e Samsung Galaxy S5/Note3 work fine, Samsung and LG lower end devices don't)

Even if I re-register. (Token is created and sent to server correctly.)

Not even the global topic sent works after reboot.

I cannot figure out what is different between install process and startup process.
Maybe a specific sequence?
Being new, not sure if I should unregister and re-register if the user starts the app.

**Comment by mnemonicflow** [comment]
@simonsmartza Have you by accident disabled notifications for the app? 
When you receive a notification and you tap on it for a few seconds, it expands and there you see a bell with a diagonal line, if you click that, notifications are disabled for that app.
Check in Settings -> Notification manager -> Filter notifications

**Comment by simonsmartza** [comment]
Nope
As explained. Notifications stops working after a reboot, only when you
re-install does it work again.
Every time the phone reboots you have to re-install

**Comment by silvolu** [comment]
@simonsmartza If I understand correctly, this is working fine on Samsung Galaxy S5/Note3, but not on other Samsung and LG devices right? What are the android's versions on the devices?

**Comment by silvolu** [comment]
Closing for inactivity, please reopen if you have more info.

**Comment by simonsmartza** [comment]
Hi
(Sorry I was away on short holiday.)

Thank you very much for your time on this!

Some more info:
On the Samsungs/LG it is dependant on localised android versions.

The Samsung S2 works with their newer Android release.
Logcat does not show me any reason why, however I might not be able (clever
enough) to pick the problem.

Some tablets and branded handsets (chinese cheaper phones for mass market)
such as those released from Vodaphone/Vodacom (South Africa) lately also
display this issue.

This is terrible for my project, as I use the cloud to send a "private" key
to the app during logon.

*They ALL display the same thing:*

*1. After installation everything works 100%.*
*2. After reboot the notifications are not received.*
*3. Re-install fixes this.*
*4. Force stopping, clearing data and restarting does not work*

I have no power over the phones and resident Android, but I am trying to
find out what happens differently during installation.
Restarting the app after forced close does not work, re-initialising
services also does not work.

My question is therefore if anyone can tell me what is different in the app
starting from an installation process.

**Comment by silvolu** [comment]
Thanks. I've opened an internal issue, I will update you here when I know more.

**Comment by silvolu** [comment]
@simonsmartza we've been trying to reproduce your issue without success. Could you generate a bug report on one of the affected devices after the reboot, save it as a file, and send it as an attachment using the [gcm support form](https://support.google.com/code/contact/gcm_dev_support)?
Also, is this affecting only messages containing the *notification* payload, or all the messages?

**Comment by simonsmartza** [comment]
Hi Silvano

I am sorry, I think I have wasted your time on this issue. I was trying to
understand what is different between installation service startup and
reboot service startup.
Also being new to Android I have been looking in the wrong place for the
problem. Apologies.

FYI,

Further testing here showed there were issues with other background
services being unstable after reboot too, possibly related (UBER app had
stability problems as well as Financial applications using OAUTH).

We have raised this issue with the actual networks distributing these
devices locally, as I think it is related to the specific Android &
Firmware combination built for those devices.
They all originate from Chinese factories, there must be a stock firmware
floating around with a bug in it.

The networks here (South Africa) have technical teams who test & approve
new phones and they have direct access to the manufacturers, they are
alerted to these now.

The flood of cheap Smartphone devices even from the likes of Samsung with
all these bugs (i.e. battery drainage, unstable GPS etc etc) into the
exploding African market is worrying for us developing VAS applications.
Almost like a sell now, fix bugs later culture :)

From our side we will use the retailers to pressurise the networks, to
pressurise the manufacturers into getting an decent Android & Firmware onto
the devices.

If the problem persists on a large scale, I will let you know and also ship
some of these devices to you to have fun with.

Regards,

Hendrik

Technical Diretor
Simon Smart ZA PtyLtd
